core/src/test/java/org/elasticsearch/action/IndicesRequestIT.java

```diff
@@ -223,6 +228,18 @@ public void testDelete() {
         assertSameIndices(deleteRequest, deleteShardActions);
     }
 
+    // for delete index throws an IndexNotFoundException if the index does not exist or create the index if external version is used
+    public void testDeleteAction() throws InterruptedException, ExecutionException {
```

Comment:
I think this is the wrong place for this - this suite checks that index names are passed correctly to sub request. IMO you should create a unit test  `TransportBulkActionTests` class, which inherits from `ESTestCase` and test how the `TransportBulkAction` decides which indices need to be auto created. You can look at how `TransportBulkActionIngestTests.TestTransportBulkAction` is set up as an example. Let me know if you need more guidance and I'm happy to help